BookShelfBusinessLogic/LibraryService.cs

```diff
@@ -0,0 +1,375 @@
+            BookAuthor similarEntry = _db.BookAuthor.FirstOrDefault(ba => ba.BookRefId == bookId && ba.AuthorRefId == authorId);
+            if (similarEntry != null)
+            {
+                return false;
+            }
+
+            _db.BookAuthor.Add(bookAuthor);
+            _db.SaveChanges();
+
+            return true;
+        }
+
+        /// <summary>
+        /// Tries to remove genre from the specified book.
+        /// </summary>
+        /// <param name="bookId"> Id of book to be updated </param>
+        /// <param name="genreId"> Id of genre to be removed </param>
+        /// <returns> True if updated, false if such genre has not been added to this book previously </returns>
+        public bool IsGenreFromBookRemoved(int bookId, int genreId)
+        {
+            BookGenre bookGenreToRemove = _db.BookGenre.FirstOrDefault(bg => bg.BookRefId == bookId && bg.GenreRefId == genreId);
+
+            if (bookGenreToRemove == null)
+            {
+                return false;
+            }
+
+            _db.BookGenre.Remove(bookGenreToRemove);
+            _db.SaveChanges();
+
+            return true;
+        }
+
+
+        /// <summary>
+        /// Tries to remove author from the specified book.
+        /// </summary>
+        /// <param name="bookId"> Id of book to be updated </param>
+        /// <param name="authorId"> Id of author to be removed </param>
+        /// <returns> True if updated, false if such author has not been added to this book previously </returns>
+        public bool IsAuthorFromBookRemoved(int bookId, int authorId)
+        {
+            BookAuthor bookAuthorToRemove = _db.BookAuthor.FirstOrDefault(ba => ba.BookRefId == bookId && ba.AuthorRefId == authorId);
+
+            if (bookAuthorToRemove == null)
+            {
+                return false;
+            }
+
+            _db.BookAuthor.Remove(bookAuthorToRemove);
+            _db.SaveChanges();
+
+            return true;
+        }
+
+        /// <summary>
+        /// Selects all Books by specified author
+        /// </summary>
+        /// <param name="authorId">Id of author</param>
+        /// <returns>The collection of Books </returns>
+        public IEnumerable<Book> GetBooksByAuthor(int authorId)
+        {
+            IEnumerable<Book> booksByAuthor = from book in _db.Books
+                                 join entry in _db.BookAuthor on book.Id equals entry.BookRefId
+                                 where entry.AuthorRefId == authorId
+                                 select book;
+
+            return booksByAuthor;
+        }
+
+        /// <summary>
+        /// Selects all Books by specified genre
+        /// </summary>
+        /// <param name="genreId">Id of genre</param>
+        /// <returns>The collection of Books </returns>
+        public IEnumerable<Book> GetBooksByGenre(int genreId)
+        {
+            IEnumerable<Book> booksByGenre = from book in _db.Books
```

Comment:
Or even write like this
![image](https://user-images.githubusercontent.com/30385787/47807156-0cf99d00-dd44-11e8-8f9f-535aaa8d0d8d.png)

Reply:
@smoukiDev personally I prefer one line join statement. it is easier to read.